### ISSUE TITLE
[FEATURE REQUEST] Support mangle properties.

### ISSUE THREAD

**Issue by finscn** [javascript-obfuscator/javascript-obfuscator#270]
the new UglifyJS supports  mangle properties.
But it loses some good feature (e.g. random mangle name )

I hope javascript-obfuscator could also  support this feature .

Thanks

**Comment by sanex3339** [comment]
It's unsafe

**Comment by finscn** [comment]
@sanex3339  I know . 
So it's an optional . And in UglifyJS , it won't mangle the common/builtin object's properties in  browser/node, and could protect the properties in `'...' / "..."` . And user could set which properties WILL /WON'T be mangled.

**Comment by srguiwiz** [comment]
One helpful feature in uglify.js is the debug option of --mangle-props.

Also, keep track of random names, not to reuse any. Use list of "names not to mangle", a user defined list and possibly a default list. Use that list also to never accept any of those names as random generated names, and list of other words not to accept (e.g. JavaScript reserved keywords). If a random generated name fails acceptance test, then try another random generation to get an acceptable random name.

**Comment by TorRanfelt** [comment]
This would be great.

**Comment by sanex3339** [comment]
Ok. I wIll try this, because the basic idea is very simple, but in most cases it probably will break the code.

The main problem - where to take the reserved properties list?
I found this list from `terser`:
https://github.com/terser/terser/blob/master/tools/domprops.js

But I don't want to add `terser` as dependency for this list only.

**Comment by sanex3339** [comment]
Base cases that should work:
```
const foo = {
    prop1: {
        prop2: 'bar'
    }
};

foo.prop1.prop2;

const {prop1} = foo;
console.log(prop1.prop2);

class Foo {
    prop1 () {}
}

const s = new Foo();

s.prop1();
```

**Comment by sanex3339** [comment]
All work here:
https://github.com/javascript-obfuscator/javascript-obfuscator/pull/625

The prototype works in some way, needs testing.
Currently, it always works (I will add an option for it later).

Who can test this?

**Comment by srguiwiz** [comment]
If you release it with this feature off by default, I'll increase the version number in our package.json, phase it in, try it out.  Won't promise quick turnaround, but will be done.  To keep your "reputation for quality" say in the doc "experimental, may still break, please report problems".  Good effort, thanks!

**Comment by sanex3339** [comment]
Now everyone can try this branch.
The option called `renameProperties`.

**Comment by sanex3339** [comment]
`renameProperties` option released as `1.1.0` version.
